web/static/js/components/respondents/RespondentIndex.jsx

```diff
@@ -453,15 +449,63 @@ class RespondentIndex extends Component<Props, State> {
     )
   }
 
+  renderHeader({ displayText, type, key, sortable, dataType }) {
+    const { sortBy, sortAsc } = this.props
+    let className = classNames({
+      thNumber: dataType === 'number',
+      thDate: dataType === 'date'
+    })
+
+    if (sortable) {
+      return (
+        <SortableHeader
+          text={displayText}
+          property={key}
+          sortBy={sortBy}
+          sortAsc={sortAsc}
+          onClick={() => this.sortBy(key)}
+          key={fieldUniqueKey(type, key)}
+          className={className}
+        />
+      )
+    } else {
+      return (
+        <th className={className} key={fieldUniqueKey(type, key)}>
+          {displayText}
+        </th>
+      )
+    }
+  }
+
+  toogleColumn(type, key) {
+    this.props.actions.toogleSelectedColumn(type, key)
+  }
+
+  columnsMenuOption({ type, key, displayText, checked }) {
+    const { selectedFields } = this.props
+    const inputId = `toggle_column_${fieldUniqueKey(type, key)}`
+    return (
+      <DropdownDropdownCheckboxItem
+        key={Math.random().toString()}
```

Comment:
keys should never be random. It will trigger a recreation of the element on each render.

Reply:
I know `key` isn't intended to be used this way. But this is the only way I found to force the component update. If I remove this line the component (and so the user selection) doesn't update properly.

It's related to [this unconventional solution](https://github.com/instedd/surveda/blob/0e0d0d2a8649415a0191f0d8a7c4491dfd1cd3ad/web/static/js/components/ui/Dropdown.jsx#L65-L73) to prevent the menu to be closed on every user selection.

Could we look for a better solution together, please? 🙏